File: src/main/java/one/devos/nautical/SofterPastels/utils/CandyTooltipItem.java
package one.devos.nautical.SofterPastels.utils;

import net.minecraft.network.chat.Component;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.TooltipFlag;
import net.minecraft.world.item.alchemy.PotionUtils;
import net.minecraft.world.level.Level;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public class CandyTooltipItem extends Item {
    public final float durationFactor;
    public CandyTooltipItem(Properties properties, float durationFactor) {
        super(properties);
        this.durationFactor = 1.0F;
    }

    @Override
    public void appendHoverText(ItemStack itemStack, @Nullable Level level, List<Component> list, TooltipFlag tooltipFlag) {
        list.add(Component.translatable("item.softerpastels.candies.hunger.description"));
        PotionUtils.addPotionTooltip(itemStack, list, 1.0F, level == null ? 20.0F : level.tickRateManager().tickrate());
        list.add(Component.translatable("item.softerpastels.candies.hunger.alwayseat"));
    }
}

File: src/main/kotlin/one/devos/nautical/SofterPastels/utils/CandyTooltipItem.kt
package one.devos.nautical.SofterPastels.utils

import net.minecraft.network.chat.Component
import net.minecraft.world.effect.MobEffectInstance
import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.TooltipFlag
import net.minecraft.world.item.alchemy.PotionUtils
import net.minecraft.world.level.Level
import java.util.*

class CandyTooltipItem(properties: Properties) : Item(properties) {

    override fun appendHoverText(itemStack: ItemStack, level: Level?, list: MutableList<Component>, tooltipFlag: TooltipFlag) {
        list.add(Component.translatable("item.softerpastels.candies.hunger.description"))

        val effects = LinkedList<MobEffectInstance>()

        if (itemStack.foodComponent != null) {
            for (effect in itemStack.foodComponent!!.effects) {
                effects.add(effect.first)
            }
        }

        PotionUtils.addPotionTooltip(effects, list, 1.0f, level?.tickRateManager()?.tickrate() ?: 20.0f)
        list.add(Component.translatable("item.softerpastels.candies.hunger.alwayseat"))
    }
}
